### PR TITLE
Transaction multichain multiple networks

### PR DESCRIPTION
**Pull request by shanejonas**
### `@metamask/package-a`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

### `@metamask/package-b`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionControllerIntegration.test.ts [hunk 1]
```diff
@@ -496,6 +503,335 @@ describe('TransactionController Integration', () => {
         );
         clock.restore();
       });
+      it.only('should be able to send and confirm transactions on different chains', async () => {
```
Comment (adonesky1) [inline packages/transaction-controller/src/TransactionControllerIntegration.test.ts:506]:
```suggestion
      it('should be able to send and confirm transactions on different chains', async () => {
```

File: packages/transaction-controller/src/TransactionControllerIntegration.test.ts [hunk 2]
```diff
@@ -496,6 +503,335 @@ describe('TransactionController Integration', () => {
...
+        await advanceTime({ clock, duration: 1 });
+        await advanceTime({ clock, duration: 1 });
+
+        expect(transactionController.state.transactions).toHaveLength(2);
+        expect(transactionController.state.transactions[0].status).toBe(
+          'confirmed',
+        );
+        expect(transactionController.state.transactions[1].status).toBe(
+          'confirmed',
+        );
```
Comment (adonesky1) [inline packages/transaction-controller/src/TransactionControllerIntegration.test.ts:832]: [nit] could we assert that these were executed on the correct chainIds/networkClients?

File: packages/transaction-controller/src/TransactionControllerIntegration.test.ts [hunk 3]
```diff
@@ -81,6 +87,7 @@ const newController = async (options: any) => {
       networkController.getNetworkClientById.bind(networkController),
     getNetworkState: () => networkController.state,
     getSelectedAddress: () => '0xdeadbeef',
```
Comment (jiexi) [inline packages/transaction-controller/src/TransactionControllerIntegration.test.ts:89]: should this just be ACCOUNT_MOCK too then?

**Comment by jiexi** [comment]
Not strictly necessary, but would be nice if we could organize the mocked requests like I have for the other scenarios, also to change the eth_getCode call to return `0x` too (which would require sendRawTransaction param value to be updates as well). Again, not strictly necessary, but would be nice since the rest are already organized and documented

**Review by adonesky1 — APPROVED** [review]
LGTM

File: packages/transaction-controller/src/TransactionControllerIntegration.test.ts [hunk 4]
```diff
@@ -496,6 +507,340 @@ describe('TransactionController Integration', () => {
...
+        expect(
+          transactionController.state.transactions[0].networkClientId,
+        ).toBe('goerli');
+        expect(transactionController.state.transactions[1].status).toBe(
+          'confirmed',
+        );
+        expect(
+          transactionController.state.transactions[1].networkClientId,
+        ).toBe('sepolia');
+        clock.restore();
```
Comment (adonesky1) [inline packages/transaction-controller/src/TransactionControllerIntegration.test.ts:842]: Need to call `stopTrackingByNetworkClientId` with `sepolia` and `goerli`

File: packages/transaction-controller/src/TransactionControllerIntegration.test.ts [hunk 5]
```diff
@@ -496,6 +507,340 @@ describe('TransactionController Integration', () => {
...
+        );
+        expect(
+          transactionController.state.transactions[0].networkClientId,
+        ).toBe('goerli');
+        expect(transactionController.state.transactions[1].status).toBe(
+          'confirmed',
+        );
+        expect(
+          transactionController.state.transactions[1].networkClientId,
+        ).toBe('sepolia');
```
Comment (adonesky1) [inline packages/transaction-controller/src/TransactionControllerIntegration.test.ts:841]: transaction orders are [reversed in trimTransactionsForState](https://github.com/MetaMask/core/blob/f4b889af485d154630e506f05a8f232f648ed16a/packages/transaction-controller/src/TransactionController.ts#L2188)

Comment (adonesky1) [reply packages/transaction-controller/src/TransactionControllerIntegration.test.ts:841]: maybe we assert that the transactions array contains the expected objects?

**Pull request merged**